runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -802,33 +787,39 @@ protected void unmapSecondaryIndexes(K key, V value) {
      */
     @SuppressWarnings("unchecked")
     protected void mapSecondaryIndexes(K key, V value) {
+        if (value == null) {
+            return;
+        }
+
         try {
-            if (value != null) {
-                // Map entry into secondary indexes
-                for (Index<K, V, ? extends Comparable> index : indexSpec) {
-                    String indexName = index.getName().get();
-                    Map<Comparable, Map<K, V>> secondaryIndex = secondaryIndexes.get(indexName);
-                    if (index.isMonoIndex()) {
-                        Comparable indexKey = index.getIndexFunction().apply(key, value);
-                        Map<K, V> slot = secondaryIndex.computeIfAbsent(indexKey, k -> new HashMap<>());
-                        slot.put(key, value);
-                    } else {
-                        for (Comparable<?> indexKey
-                                : index.getMultiValueIndexFunction().apply(key, value)) {
-                            Map<K, V> slot = secondaryIndex
-                                    .computeIfAbsent(indexKey, k -> new HashMap<>());
-                            slot.put(key, value);
-                        }
-                    }
+            // Map entry into secondary indexes
```

Comment:
this method and the prev. one looks like a copy-paste. Is it possible to get rid of copy paste here?